atomic_hooks/Cargo.toml

```diff
@@ -7,11 +7,11 @@ description = "Atomic Reative State and Local State Hooks"
 license = "Apache-2.0 OR MIT"
 
 [dependencies]
-slotmap = "0.4.0"
-anymap = "0.12.1" 
-illicit = "1.1.0"
-typemap = "0.3.3"
+slotmap = "^1.0.6"
```

Comment:
Why we need carets (`^`)? (The question applies to all deps.)

> The string "0.1.12" is a semver version requirement. Since this string does not have any operators in it, it is interpreted the same way as if we had specified "^0.1.12"
- https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Reply:
This is what we normally do. What is desired here?

Reply:
I would use versions without carets to make it consistent and if I'm not wrong it's standard among Rust repos.

(However I understand it could be confusing - I would assume `1.2.3` is `=1.2.3` and not `^1.2.3`. But I'd rather follow conventions, because it's a Rust open-source project.)

Reply:
Changed